crates/autopilot/src/periodic_db_cleanup.rs

```diff
@@ -0,0 +1,170 @@
+            match self.db.delete_order_events_before(timestamp).await {
+                Ok(affected_rows_count) => {
+                    tracing::debug!(affected_rows_count, timestamp = %timestamp.to_string(), "order events cleanup");
+                    Metrics::get().order_events_cleanup_total.inc()
+                }
+                Err(err) => {
+                    tracing::warn!(?err, "failed to delete order events before {}", timestamp)
+                }
+            }
+            interval.tick().await;
+        }
+    }
+}
+
+#[derive(prometheus_metric_storage::MetricStorage)]
+struct Metrics {
+    /// The total number of successful `order_events` table cleanups
+    #[metric(name = "periodic_db_cleanup")]
+    order_events_cleanup_total: prometheus::IntCounter,
+}
+
+impl Metrics {
+    fn get() -> &'static Self {
+        Metrics::instance(observe::metrics::get_storage_registry()).unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        database::{
+            byte_array::ByteArray,
+            order_events::{OrderEvent, OrderEventLabel},
+        },
+        itertools::Itertools,
+        sqlx::{PgPool, Row},
+    };
+
+    #[tokio::test]
+    #[ignore]
+    async fn order_events_cleaner_flow() {
+        let db = Postgres::new("postgresql://").await.unwrap();
+        let mut ex = db.0.begin().await.unwrap();
+        database::clear_DANGER_(&mut ex).await.unwrap();
+
+        let now = Utc::now();
+        let event_a = OrderEvent {
+            order_uid: ByteArray([1; 56]),
+            timestamp: now - chrono::Duration::milliseconds(3000),
+            label: OrderEventLabel::Created,
+        };
+        let event_b = OrderEvent {
+            order_uid: ByteArray([2; 56]),
+            timestamp: now - chrono::Duration::milliseconds(1000),
+            label: OrderEventLabel::Created,
+        };
+        let event_c = OrderEvent {
+            order_uid: ByteArray([3; 56]),
+            timestamp: now,
+            label: OrderEventLabel::Created,
+        };
+
+        database::order_events::insert_order_event(&mut ex, &event_a)
+            .await
+            .unwrap();
+        database::order_events::insert_order_event(&mut ex, &event_b)
+            .await
+            .unwrap();
+        database::order_events::insert_order_event(&mut ex, &event_c)
+            .await
+            .unwrap();
+
+        ex.commit().await.unwrap();
+
+        let ids = order_event_ids_before(&db.0).await;
+        assert_eq!(ids.len(), 3);
+        assert!(ids.contains(&event_a.order_uid));
+        assert!(ids.contains(&event_b.order_uid));
+        assert!(ids.contains(&event_c.order_uid));
+
+        let config =
+            OrderEventsCleanerConfig::new(Duration::from_millis(500), Duration::from_millis(2000));
+        let cleaner = OrderEventsCleaner::new(config, db.clone());
+
+        tokio::task::spawn(cleaner.run_forever());
```

Comment:
I am personally against the `fire and forget` pattern. I believe the thread will continue executions in the background during the execution of other tests, which could lead to unexpected results. Currently, it works because `order_events` table is not queried in any other test. One more reason for the https://github.com/cowprotocol/services/pull/2023